packages/hint-no-disallowed-headers/src/hint.ts

```diff
@@ -43,75 +94,53 @@ export default class NoDisallowedHeadersHint implements IHint {
             'x-version'
         ];
 
+        // These headers are disallowed for specific reasons.
+        const specialDisallowedHeaders: ISpecialHeadersWithMessage = {
+            expires: 'disallowedExpiresHeader',
+            host: 'disallowedHostHeader',
+            p3p: 'disallowedP3PHeader',
+            pragma: 'disallowedPragmaHeader',
+            via: 'disallowedViaHeader',
+            'x-frame-options': 'disallowedXFrameOptionsHeader'
+        };
+
         let includeHeaders: string[];
         let ignoreHeaders: string[];
 
+        /*
+         * Augment the list of generally disallowed headers with those configured in options.include, and remove those
+         * configured in options.ignore.
+         */
         const loadHintConfigs = () => {
             includeHeaders = (context.hintOptions && context.hintOptions.include) || [];
             ignoreHeaders = (context.hintOptions && context.hintOptions.ignore) || [];
 
-            disallowedHeaders = mergeIgnoreIncludeArrays(disallowedHeaders, ignoreHeaders, includeHeaders);
+            generalDisallowedHeaders = mergeIgnoreIncludeArrays(generalDisallowedHeaders, ignoreHeaders, includeHeaders);
         };
 
-        const serverHeaderContainsTooMuchInformation = (serverHeaderValue: string): boolean => {
-
-            const regex = [
-                /*
-                 * Version numbers.
-                 *
-                 * e.g.:
-                 *
-                 *   apache/2.2.24 (unix) mod_ssl/2.2.24 openssl/1.0.1e-fips mod_fastcgi/2.4.6
-                 *   marrakesh 1.9.9
-                 *   omniture dc/2.0.0
-                 *   microsoft-iis/8.5
-                 *   pingmatch/v2.0.30-165-g51bed16#rel-ec2-master i-077d449239c04b184@us-west-2b@dxedge-app_us-west-2_prod_asg
-                 */
-
-                /\/?v?\d\.(\d+\.?)*/,
-
-                /*
-                 * OS/platforms names (usually enclose between parentheses).
-                 *
-                 * e.g.:
-                 *
-                 *  apache/2.2.24 (unix) mod_ssl/2.2.24 openssl/1.0.1e-fips mod_fastcgi/2.4.6
-                 *  apache/2.2.34 (amazon)
-                 *  nginx/1.4.6 (ubuntu)
-                 */
-
-                /\(.*\)/,
-
-                /*
-                 * Compiled-in modules.
-                 *
-                 * e.g.:
-                 *
-                 *  apache/2.2.24 (unix) mod_ssl/2.2.24 openssl/1.0.1e-fips mod_fastcgi/2.4.6
-                 *  apache/2.4.6 (centos) php/5.4.16
-                 *  jino.ru/mod_pizza
-                 */
-
-                /(mod_|openssl|php)/
-            ];
-
-            return regex.some((r) => {
-                return r.test(serverHeaderValue);
-            });
-        };
+        const validateGeneralHeaders = ({ response, resource }: FetchEnd) => {
+            const headers: string[] = includedHeaders(response.headers, generalDisallowedHeaders);
```

Comment:
You don't actually need to change this since it was pre-existing code you moved, but I wanted to call out we've been slowly moving toward dropping redundant types in our existing code when we make updates. Keeps things a bit easier to read and closer to JS. Probably worth us finding/creating a linting rule at some point to track down the rest of these...
```suggestion
            const headers = includedHeaders(response.headers, generalDisallowedHeaders);
```

Reply:
Thanks for letting me know. I'll do this change since it's so simple. And I'll keep that in mind for the future.